libraries/SuplaDevice/supla/network/esp_wifi.h

```diff
@@ -107,19 +152,34 @@ class ESPWifi : public Supla::Network {
         [](const WiFiEventStationModeDisconnected &event) {
           Serial.println("wifi Station disconnected");
         });
+	
+#ifdef ESP_SSL
+     
+	if (fingerprint)
+      client.setFingerprint(fingerprint);
+	else
+	  client.setInsecure();
+	
+#endif
 
     Serial.print("WIFI: establishing connection with SSID: \"");
     Serial.print(ssid);
     Serial.println("\"");
+
     WiFi.begin(ssid, password);
-    yield();
+	yield();
   }
-
  protected:
-  WiFiClient client;
-
-  char ssid[MAX_SSID_SIZE];
-  char password[MAX_WIFI_PASSWORD_SIZE];
+#ifdef ESP_SSL
+   WiFiClientSecure client;
+#else
+   WiFiClient client;
+#endif
```

Comment:
WiFiClientSecure inherits from WiFiClient, so you can keep pointer to WiFiClient and create proper instance in connect() methods (if client==nullptr then create instance)

Reply:
Yes, but there is an unwritten rule to declare objects statically and not dynamically in arduino projects. This is due to checking the amount of memory needed at compile time. This may cause a valid build, but the crash when trying to run. Your call -> I can change it but remember this.

Reply:
this is anyway for ESP board where there is plenty of memory ;)